internal/pkg/client/ocisif/auth.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Contributors to the Apptainer project, established as
+//   Apptainer a Series of LF Projects LLC.
+//   For website terms of use, trademark policy, privacy policy and other
+//   project policies see https://lfprojects.org/policies
+// Copyright (c) 2020, Control Command Inc. All rights reserved.
+// Copyright (c) 2020-2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package ocisif
```

Comment:
I don't think we should be adding any ocisif.  Is this required by any of our code?

Reply:
I see the AuthOptn function is used elsewhere, but maybe we should just be calling the package something else.

Reply:
Can we just move this file to `internal/pkg/client/oras/auth.go` ?

Reply:
Also, please be careful with the copyright...

The auth keychain stuff contains 3rd party code from Google. See how this is mentioned in our repo here:

https://github.com/sylabs/singularity/blob/8c1e06338f5ada6c17403bc6899c6d8742317cd7/internal/pkg/remote/credential/ociauth/ociauth.go#L106

On this PR, that information has been dropped from the header in Apptainer:

https://github.com/apptainer/apptainer/pull/1866/files#diff-aead9acf55ec4678e340ed22d8da1139dcd08dcd3bfca4ad816e3ea1743498e8